### PR TITLE
Implement batchLimit on PersistentHitQueue

### PR DESCRIPTION
**Pull request by addb**
## Description

Added batchLimit to the PersistentHitQueue so the hits would be processed only after the queue size reaches batchLimit.

## Motivation and Context

Batching is required for Analytics hits.

## How Has This Been Tested?

Have added Unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift [hunk 1]
```diff
@@ -19,13 +19,17 @@ public class PersistentHitQueue: HitQueuing {
     private static let DEFAULT_RETRY_INTERVAL = TimeInterval(30)
     private var suspended = true
     private let queue = DispatchQueue(label: "com.adobe.mobile.persistenthitqueue")
+    private var batchLimit = 0
```
Comment (nporter-adbe) [inline AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:22]: My opinion is that the batchLimit should be defined on the `HitProcessing` rather than directly injected into the persistent hit queue. This is so that other uses of HitProcessing can leverage the batch limit, and allows each HitProcessing to decide if they want a setter on their batch limit or an immutable batch limit.

Comment (addb) [reply AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:22]: Yes that makes sense.

File: AEPIdentity/Sources/IdentityHitProcessor.swift [hunk 2]
```diff
@@ -27,6 +27,7 @@ class IdentityHitProcessor: HitProcessing {
     }
 
     // MARK: HitProcessing
+    var batchLimit = 0
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/IdentityHitProcessor.swift:30]: If this doesn't change anywhere, lets me it a `let`.

File: AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift [hunk 3]
```diff
@@ -19,10 +19,13 @@ public class PersistentHitQueue: HitQueuing {
     private static let DEFAULT_RETRY_INTERVAL = TimeInterval(30)
     private var suspended = true
     private let queue = DispatchQueue(label: "com.adobe.mobile.persistenthitqueue")
+    private var currentBatchSize = 0
+    private var batching = false
 
     /// Creates a new `HitQueue` with the underlying `DataQueue` which is used to persist hits
     /// - Parameter dataQueue: a `DataQueue` used to persist hits
     /// - Parameter processor: a `HitProcessing` used to process hits
+    /// - Parameter batchLimit: an `Int` used to set minimum number of queued hits to start processing hits
```
Comment (nporter-adbe) [inline AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:28]: Looks like this comment can be removed.

File: AEPSignal/Sources/SignalHitProcessor.swift [hunk 4]
```diff
@@ -21,6 +21,7 @@ class SignalHitProcessor: HitProcessing {
     }
 
     // MARK: - HitProcessing
+    var batchLimit = 0
```
Comment (nporter-adbe) [inline AEPSignal/Sources/SignalHitProcessor.swift:24]: `let`

File: AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift [hunk 5]
```diff
@@ -58,8 +69,16 @@ public class PersistentHitQueue: HitQueuing {
     }
 
     /// A recursive function for processing hits, it will continue processing all the hits until none are left in the data queue
-    private func processNextHit() {
+    /// - Parameter ignoreBatchLimit: a `Bool` flag to determine batching hits
+    private func processNextHit(ignoreBatchLimit: Bool = false) {
         queue.async {
+            if self.batching && !ignoreBatchLimit {
```
Comment (nporter-adbe) [inline AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:75]: Can you add a few comments in English describing this logic? I find it a bit hard to follow just reading it.

**Review by nporter-adbe — APPROVED** [review]
